src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -160,7 +161,12 @@ public IReadOnlyList<TagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotS
                         // tags. Only dealing with the dataTagSpans if the count is 1 because we do not see a multi-buffer case
                         // occurring
                         var dataTagSpans = tag.Span.GetSpans(snapshot);
-                        if (dataTagSpans.Count == 1)
+
+                        // Checks to see if location and hint name are already present in the cache.
+                        var cacheContainsTag = _cache.Any(c => (c.mappingTagSpan.Tag.Hint.Span == tag.Tag.Hint.Span)
+                            && (c.mappingTagSpan.Tag.Hint.DisplayParts[0] == tag.Tag.Hint.DisplayParts[0]));
```

Comment:
1. why the display part check?
2. this is n^2 in cost.  move to some sort of pooled set/dictionary to make sure that we cna do this in linear time.

Reply:
is there ever any case where two tags should occupy the same position?  if not, just check that hte hint span's start is already taken.